src/Compilers/Core/Portable/SourceGeneration/Nodes/SharedInputNodes.cs

```diff
@@ -24,5 +24,7 @@ internal static class SharedInputNodes
         public static readonly InputNode<SyntaxTree> SyntaxTrees = new InputNode<SyntaxTree>(b => b.Compilation.SyntaxTrees.ToImmutableArray());
 
         public static readonly InputNode<AnalyzerConfigOptionsProvider> AnalyzerConfigOptions = new InputNode<AnalyzerConfigOptionsProvider>(b => ImmutableArray.Create(b.DriverState.OptionsProvider));
+
+        public static readonly InputNode<MetadataReference> MetadataReferences = new InputNode<MetadataReference>(b => b.Compilation.References.AsImmutable());
```

Comment:
As a bit of an aside Im' surprised this already isn't an `ImmutableArray`. The implementation is an iterator and we use it several times during compilation.